class.jetpack-gutenberg.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+/**
+ * General Gutenberg editor specific functionality
+ */
+class Jetpack_Gutenberg {
+	/**
+	 * Check if Gutenberg editor is available
+	 *
+	 * @since 6.7.0
+	 *
+	 * @return bool
+	 */
+	public static function is_gutenberg_available() {
+		return function_exists( 'register_block_type' );
+	}
+
+	/**
+	 * Load Gutenberg assets
+	 *
+	 * @since 6.7.0
+	 *
+	 * @return void
+	 */
+	public static function enqueue_block_assets() {
+		if ( ! self::should_load_blocks() ) {
+			return;
+		}
+
+		$rtl = is_rtl() ? '.rtl' : '';
+
+		/**
+		 * Filter to enable serving blocks via CDN
+		 *
+		 * CDN cache is busted once a day or when Jetpack version changes. To customize it:
+		 *   add_filter( 'jetpack_gutenberg_cdn_cache_buster', function( $version ) { return time(); }, 10, 1 );
+		 *
+		 * @since 6.5.0
+		 *
+		 * @param bool false Whether to load Gutenberg blocks from CDN
+		 */
+		if ( apply_filters( 'jetpack_gutenberg_cdn', false ) ) {
+			$cdn_base    = 'https://s0.wp.com/wp-content/mu-plugins/jetpack/_inc/blocks';
+			$view_script = "$cdn_base/view.js";
+			$view_style  = "$cdn_base/view$rtl.css";
+
+			/**
+			 * Filter to modify cache busting for Gutenberg block assets loaded from CDN
+			 *
+			 * @since 6.5.0
+			 *
+			 * @param string
+			 */
+			$version = apply_filters( 'jetpack_gutenberg_cdn_cache_buster', sprintf( '%s-%s', gmdate( 'd-m-Y' ), JETPACK__VERSION ) );
+		} else {
+			$view_script = plugins_url( '_inc/blocks/view.js', JETPACK__PLUGIN_FILE );
+			$view_style  = plugins_url( "_inc/blocks/view$rtl.css", JETPACK__PLUGIN_FILE );
+			$version     = Jetpack::is_development_version() && file_exists( JETPACK__PLUGIN_DIR . '_inc/blocks/view.js' )
+				? filemtime( JETPACK__PLUGIN_DIR . '_inc/blocks/view.js' )
+				: JETPACK__VERSION;
+		}
+
+		wp_enqueue_script( 'jetpack-blocks-view', $view_script, array(), $version );
+		wp_enqueue_style( 'jetpack-blocks-view', $view_style, array(), $version );
+		wp_add_inline_script(
+			'wp-i18n',
+			'wp.i18n.setLocaleData( ' . Jetpack::get_i18n_data_json() . ', \'jetpack\' );'
+		);
```

Comment:
This is a bit surprising to see here because `wp-i18n` is not a dependency of the script that is enqueued here 😕 

In practice, that's fine because this action will always run when the editor action is run (which does enqueue scripts that depend on `wp-i18n`. Effectively, it's the same. Leaving it here means this it won't be enqueued until it's required.

Additionally, I expect the view scripts to depend on `wp-i18n` in the near future (#10319).

Let's leave it as it.

Reply:
This was added with the VR block.

Asked about these same lines here: https://github.com/Automattic/jetpack/pull/10331#issuecomment-431139029

Leaving sounds good, perhaps we should add a note that they aren't used anywhere (until we actually do use them)?

Reply:
🤔 Ok, I've rethought this.

Until we actually need i18n here, we shouldn't do this. If some other frontend script is requiring `wp-i18n`, we'll be adding our Jetpack locale data without need.

That's good in the sense that it "falls through" to the editor when we depend on it, but bad in the sense that our view doesn't depend on it but we may be adding it unnecessarily to on the frontend.

For now, I'm going to move it to the editor section and add a note in #10319.

Reply:
FWIW, we're relying on this piece in order to be able to reuse Jetpack translations in Gutenberg blocks (cc @ockham)